### PR TITLE
clean way of skipping expensive and integration tests. fix #1155

### LINKED ISSUE #1155: exclusion of heavy/long/dependency requiring tests
1) the tests in stacktest all require docker and docker-compose to run a stack. they also require `make bin` to have run.
2) `idx/cassandra` has a few Benchmarks that CreateSession and require a cassandra to be running.

these tests are:
A) time consuming to always run
B) not trivial to get the context right. For 2 we could probably just use docker also though, and any dependency can basically solved this way, so that only problem 1 remains.

As for how to exclude tests, there seem to be 2 common approaches:
a) build tags: see https://stackoverflow.com/questions/24030059/skip-some-tests-with-go-test and https://medium.com/@povilasve/go-advanced-tips-tricks-a872503ac859
b) env vars: see https://peter.bourgon.org/go-for-industrial-programming/#testing, https://splice.com/blog/lesser-known-features-go-test/
c) short mode: https://golang.org/pkg/testing/#Short

some preliminary pros/cons after not much research:
- b and c require more setup than a, which is mostly implicit.
- a only on file level, whereas b is per function
- b and c can provide feedback back to user, when `-v` is used. a has no feedback mechanism afaik.
- according to peter bourgon's best practices long tests should be opt-in, which would rule out c. However, if the go developers made the short flag as to opt-out, then I rather go with how go is actually designed.

I think i would prefer c.

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

File: stacktest/tests/end2end_carbon_bigtable/end2end_carbon_test.go [hunk 1]
```diff
@@ -31,6 +33,11 @@ func init() {
 	log.SetLevel(log.InfoLevel)
 }
 func TestMain(m *testing.M) {
+	flag.Parse()
+	if testing.Short() {
+		fmt.Println("skipping test in short mode")
```
Comment (robert-milan) [inline stacktest/tests/end2end_carbon_bigtable/end2end_carbon_test.go]:
```suggestion
		fmt.Println("skipping end2end carbon test (bigtable) in short mode")
```

File: idx/cassandra/cassandra_test.go [hunk 2]
```diff
@@ -419,6 +419,9 @@ func TestFind(t *testing.T) {
 }
 
 func BenchmarkIndexing(b *testing.B) {
+	if testing.Short() {
+		b.Skip("skipping benchmark in short mode")
```
[the remark this answers is not in the document]
Comment (Dieterbe) [reply idx/cassandra/cassandra_test.go]: the example in the golang guidelines don't attempt to print the name, see overview section at https://golang.org/pkg/testing/

I think the reason is that the name is always already shown anyway. e.g. output would look like:
```
--- PASS: TestFind (0.00s)
=== RUN   TestPruneStaleOnLoad
--- PASS: TestPruneStaleOnLoad (0.00s)
=== RUN   TestPruneStaleOnLoadWithTags
--- PASS: TestPruneStaleOnLoadWithTags (0.00s)
--- SKIP: BenchmarkIndexing
    cassandra_test.go:423: skipping benchmark in short mode
--- SKIP: BenchmarkLoad
    cassandra_test.go:474: skipping benchmark in short mode
--- SKIP: BenchmarkIndexingWithUpdates
    cassandra_test.go:510: skipping benchmark in short mode
```

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -10,7 +10,8 @@ Thanks for your interest in contributing to metrictank!
 
 * `make bin`: for building all binaries.
 * `make docker`: for building docker image (after building binaries)
-* `make test`: unit tests
+* `make test`: unit tests minus expensive integration tests
```
Comment (Dieterbe) [inline docs/CONTRIBUTING.md]:
```suggestion
* `make test`: unit tests excluding expensive integration tests (faster)
```

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -10,7 +10,8 @@ Thanks for your interest in contributing to metrictank!
...
+* `make test-all`: unit tests including expensive integration tests
```
Comment (Dieterbe) [inline docs/CONTRIBUTING.md]:
```suggestion
* `make test-all`: unit tests including expensive integration tests (slower)
```

**Pull request merged**